src/Arcus.WebApi.Telemetry.Serilog/Correlation/CorrelationInfoEnricher.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using Arcus.WebApi.Correlation;
+using GuardNet;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Serilog.Core;
+using Serilog.Events;
+
+namespace Arcus.WebApi.Telemetry.Serilog.Correlation
+{
+    /// <summary>
+    /// Enriches the log events with the correlation information.
+    /// </summary>
+    public class CorrelationInfoEnricher : ILogEventEnricher
+    {
+        private const string TransactionIdProperty = "TransactionId",
+                             OperationIdProperty = "OperationId";
+        
+        private readonly IServiceProvider _services;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CorrelationInfoEnricher"/> class.
+        /// </summary>
+        public CorrelationInfoEnricher(IServiceProvider services)
+        {
+            Guard.NotNull(services, nameof(services));
+
+            _services = services;
+        }
+
+        /// <summary>Enrich the log event.</summary>
+        /// <param name="logEvent">The log event to enrich.</param>
+        /// <param name="propertyFactory">Factory for creating new properties to add to the event.</param>
+        public void Enrich(LogEvent logEvent, ILogEventPropertyFactory propertyFactory)
+        {
+            Guard.NotNull(logEvent, nameof(logEvent));
+            Guard.NotNull(propertyFactory, nameof(propertyFactory));
+
+            var correlationInfo = _services.GetRequiredService<HttpCorrelationInfo>();
```

Comment:
Thinking out loud - Can we make this depend on `CorrelationInfo` so that we can move it to Arcus Observation eventually?

Reply:
I know. The problem is that the HTTP variant is a service and the correlation is a request feature. Maybe if we could make the correlation a scoped service (for each request). If that's possible, and at least as safe as the request features.

Reply:
Let's keep it as-is and see how we'll do that later on!